go-controller/pkg/ovn/ovn.go

```diff
@@ -605,6 +605,7 @@ func (oc *Controller) addRetryPods(pods []kapi.Pod) {
 	oc.retryPodsLock.Lock()
 	defer oc.retryPodsLock.Unlock()
 	for _, pod := range pods {
+		pod := pod
```

Comment:
@tssurya should we do the same in `initRetryPod()`? That takes a pod pointer too...

Reply:
but in `initRetryPod` we are not looping over pointers, each time the pointer pod object is passed as argument to the function so I'd think we are good there? It's called from:

```
oc.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
		AddFunc: func(obj interface{}) {
			pod := obj.(*kapi.Pod)
			go oc.metricsRecorder.AddPodEvent(pod.UID)
			oc.initRetryPod(pod)
			if !oc.ensurePod(nil, pod, true) {
				oc.unSkipRetryPod(pod)
				return
			}
			oc.checkAndDeleteRetryPod(pod.UID)
		},
```
where we are newly defining pod variable every time in the handler instead of reusing a variable.

Reply:
@tssurya I was just thinking that we don't really know what happens in ensurePod() and if it pokes fields in the pod object. We shouldn't be, but I thought it might be safer.